aea/agent_loop.py

```diff
@@ -172,59 +174,69 @@ def _behaviour_exception_callback(self, fn: Callable, exc: Exception) -> None:
         :return: None
         """
         self.logger.exception(
-            f"Loop: Exception: `{exc}` occured during `{fn}` processing"
+            f"Loop: Exception: `{exc}` occured during `{task_callable}` processing"
         )
         self._exceptions.append(exc)
         self._state.set(AgentLoopStates.error)
 
-    def _register_behaviour(self, behaviour: Behaviour) -> None:
+    def _register_periodic_task(
+        self,
+        task_callable: Callable,
+        period: float,
+        start_at: Optional[datetime.datetime],
+    ) -> None:
         """
         Register behaviour to run periodically.
 
         :param behaviour: Behaviour object
 
         :return: None
         """
-        if behaviour in self._behaviours_registry:  # pragma: nocover
+        if task_callable in self._periodic_tasks:  # pragma: nocover
             # already registered
             return
 
         periodic_caller = PeriodicCaller(
             partial(
                 self._agent._execution_control,  # pylint: disable=protected-access # TODO: refactoring!
-                behaviour.act_wrapper,
-                behaviour,
+                task_callable,
+                task_callable,
```

Comment:
why twice same arg?

Reply:
yes, one is the callable, another is a component, this should be updated later, on refactoring.